### PR TITLE
tests: add default_root_dir=tmpdir

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

some test were influenced by previous runs...

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
Not sure why MLFlow fails in CI while locally pass
```
        file_path = os.path.join(root, file_name)
        if not exists(file_path):
>           raise MissingConfigException("Yaml file '%s' does not exist." % file_path)
E           mlflow.exceptions.MissingConfigException: Yaml file '/tmp/pytest-of-runner/pytest-0/test_loggers_fit_test_MLFlowLo0/default/a7c4907de41943a7b958534289a872a2/meta.yaml' does not exist.
```

**Comment by awaelchli** [comment]
same here #2391 what's going on? I've been trying to debug this for literally almost the whole day. 
I think it is a mismatch between the logger dir (weights_save_path) and the checkpoint dir .

**Comment by Borda** [comment]
@awaelchli so  yo say that `weights_save_path` and `default_root_dir` does not like each other?

**Comment by awaelchli** [comment]
yes, the bug is somewhere burried in the many if else statements that sets these two variables. There seems to be a mismatch betwen these two paths.
Basically some loggers (e.g. mlflow) save a config file like meta.yaml in the log dir and this seems to be in the wrong folder. 
Then when you try to resume from checkpoint  (e.g. in trainer.test) it will not find that file.

**Comment by Borda** [comment]
> yes, the bug is somewhere burried in the many if else statements that sets these two variables. There seems to be a mismatch betwen these two paths.

at first I thought just to skip it and use the passing test maybe it is a chance to find the bug and fix it...

**Comment by awaelchli** [comment]
@Borda #2502 got merged, this is now unblocked

File: tests/trainer/test_trainer.py [hunk 1]
```diff
@@ -182,6 +182,17 @@ def _optimizer_step(epoch, batch_idx, optimizer, optimizer_idx,
...
+    schedule = {1: 2, 3: 4}
+
+    trainer = Trainer(
+        accumulate_grad_batches=schedule,
+        limit_train_batches=0.1,
+        limit_val_batches=0.1,
+        max_epochs=2,
+        default_root_dir=tmpdir,
+    )
+
```
Comment (awaelchli) [inline tests/trainer/test_trainer.py:195]: this redefines the trainer and schedule, and the pytest.mark.parameterize has no effect. 
Is this intentional?

**Comment by Borda** [comment]
@awaelchli thx for fixing missed changes after rebase :]

File: tests/trainer/test_trainer.py (hunk 1 above)
Comment (Borda) [reply tests/trainer/test_trainer.py]: no, this was a rebasing issue...

**Pull request merged**